### PR TITLE
Add initial arch doc.

### PR DESCRIPTION
**Pull request by robyurkowski**
WIP, but merging okay as documentation is ALWAYS WIP.

### CODE REVIEW

File: documentation/architecture.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+At least that's the theory.
+
+## A Reminder
+
+The architecture of Afterburner is quite a bit different from Rails. It may seem like you're not making progress, or like it takes twice as many files to write an extension for Afterburner as it does for a Rails app.
+
+Keep two things in mind:
+
+1. When you're learning something new, it's natural that things will take a bit longer;
+2. Afterburner shifts code out of controllers and models and into Domain Controllers and Use Cases repsectively. You might wind up creating more files, but these files will be simpler and will make it easier to add, change, test, and refactor your code in the future. So, yes, you do have to create and manage a few extra files, but the trade-off is worth it. Trust us.
```
Comment (robyurkowski) [inline documentation/architecture.md:24]: Coupla updates needed here.

Comment (knewter) [inline documentation/architecture.md:24]: respectively

Comment (robyurkowski) [inline documentation/architecture.md:24]: Fixed in my upcoming commit already ;)

**Comment by robyurkowski** [comment]
Merged.

**Pull request closed without merging**